EFT_Trigger_Helper.cls

```diff
@@ -0,0 +1,39 @@
+public with sharing class EFT_Trigger_Helper {
+	public static void createCaseWhenNeeded(List<EFT_Transaction_Status__c> eft_trans){
+		List<Case> cases = new List<Case>();
+		List<EFT_Transaction_Status__c> eft_fields = new List<EFT_Transaction_Status__c> 
```

Comment:
No need to write query in list constructor you can simply write it and assign result to list. You can opt for `SOQL for loop`